### PR TITLE
[ticket/13420] Correctly filter by lt/gt/equal to 0 posts when pruning users.

### PR DESCRIPTION
**Pull request by prototech**
[PHPBB3-13420](https://tracker.phpbb.com/browse/PHPBB3-13420)

### CODE REVIEW

**Comment by nickvergessen** [comment]
@prototech would be nice to have it ;)

File: phpBB/includes/acp/acp_prune.php [hunk 1]
```diff
@@ -397,7 +397,7 @@ function get_prune_users(&$user_ids, &$usernames)
 			$joined_after = request_var('joined_after', '');
 			$active = request_var('active', '');
 
-			$count = request_var('count', 0);
+			$count = ($request->variable('count', '') === '') ? false : $request->variable('count', 0);
```
Comment (Elsensee) [inline phpBB/includes/acp/acp_prune.php]: Why don't you just use type casting here?

Comment (nickvergessen) [inline phpBB/includes/acp/acp_prune.php]: could also try `$request->variable('count', -1)` ? and then check for `($count >= 0)`

Comment (nickvergessen) [inline phpBB/includes/acp/acp_prune.php]: okay, we use a similar mechanism already for the posts in queue:
```php
$posts_on_queue = (trim($request->variable('posts_on_queue', '')) === '') ? false : $request->variable('posts_on_queue', 0);
```
so I guess its fine?

**Pull request merged**